File: app/src/main/java/dev/ragnarok/fenrir/api/interfaces/IBoardApi.java
package dev.ragnarok.fenrir.api.interfaces;

import androidx.annotation.CheckResult;

import java.util.Collection;

import dev.ragnarok.fenrir.api.model.IAttachmentToken;
import dev.ragnarok.fenrir.api.model.response.DefaultCommentsResponse;
import dev.ragnarok.fenrir.api.model.response.TopicsResponse;
import io.reactivex.rxjava3.core.Single;


public interface IBoardApi {

    @CheckResult
    Single<DefaultCommentsResponse> getComments(int groupId, int topicId, Boolean needLikes, Integer startCommentId,
                                                Integer offset, Integer count, Boolean extended,
                                                String sort, String fields);

    @CheckResult
    Single<Boolean> restoreComment(int groupId, int topicId, int commentId);

    @CheckResult
    Single<Boolean> deleteComment(int groupId, int topicId, int commentId);

    @CheckResult
    Single<TopicsResponse> getTopics(int groupId, Collection<Integer> topicIds, Integer order,
                                     Integer offset, Integer count, Boolean extended,
                                     Integer preview, Integer previewLength, String fields);

    @CheckResult
    Single<Boolean> editComment(int groupId, int topicId, int commentId,
                                String message, Collection<IAttachmentToken> attachments);

    @CheckResult
    Single<Integer> addComment(Integer groupId, int topicId, String message,
                               Collection<IAttachmentToken> attachments, Boolean fromGroup,
                               Integer stickerId, Integer generatedUniqueId);

}

File: app_fenrir/src/main/kotlin/dev/ragnarok/fenrir/api/interfaces/IBoardApi.kt
package dev.ragnarok.fenrir.api.interfaces

import androidx.annotation.CheckResult
import dev.ragnarok.fenrir.api.model.interfaces.IAttachmentToken
import dev.ragnarok.fenrir.api.model.response.DefaultCommentsResponse
import dev.ragnarok.fenrir.api.model.response.TopicsResponse
import io.reactivex.rxjava3.core.Single

interface IBoardApi {
    @CheckResult
    fun getComments(
        groupId: Long, topicId: Int, needLikes: Boolean?, startCommentId: Int?,
        offset: Int?, count: Int?, extended: Boolean?,
        sort: String?, fields: String?
    ): Single<DefaultCommentsResponse>

    @CheckResult
    fun restoreComment(groupId: Long, topicId: Int, commentId: Int): Single<Boolean>

    @CheckResult
    fun deleteComment(groupId: Long, topicId: Int, commentId: Int): Single<Boolean>

    @CheckResult
    fun getTopics(
        groupId: Long, topicIds: Collection<Int>?, order: Int?,
        offset: Int?, count: Int?, extended: Boolean?,
        preview: Int?, previewLength: Int?, fields: String?
    ): Single<TopicsResponse>

    @CheckResult
    fun editComment(
        groupId: Long, topicId: Int, commentId: Int,
        message: String?, attachments: Collection<IAttachmentToken>?
    ): Single<Boolean>

    @CheckResult
    fun addComment(
        groupId: Long?, topicId: Int, message: String?,
        attachments: Collection<IAttachmentToken>?, fromGroup: Boolean?,
        stickerId: Int?, generatedUniqueId: Int?
    ): Single<Int>
}